velox/expression/Expr.cpp

```diff
@@ -197,14 +196,13 @@ void Expr::computeMetadata() {
     deterministic_ = vectorFunction_->isDeterministic();
   }
 
-  std::set<FieldReference*> allFields;
   for (auto& input : inputs_) {
     input->computeMetadata();
     deterministic_ &= input->deterministic_;
     propagatesNulls_ &= input->propagatesNulls_;
+    updateMultiplyReferencedFields(
```

Comment:
This code calls updateMultiplyReferencedFields which checks each field in input->distinctFields_ against distinctFields_, then calls mergeFields which again checks each field in input->distinctFields_ against distinctFields_. It seems that checking twice is unnecessary and can be eliminated by combining the two functions:

```
void mergeFields(
    std::vector<FieldReference*>& distinctFields,
    std::set<FieldReference*>& multiRefFields,
    const std::vector<FieldReference*>& moreFields) {
  for (auto* newField : moreFields) {
    if (isMember(distinctFields, *newField)) {
      multiRefFields.emplace_back(newField);
    } else {
      fields.emplace_back(newField);
    }
  }
}
```